docs/doc-system.md

```diff
@@ -0,0 +1,114 @@
+Slang Doc System 
+================
+
+Slang contains a rudimentary documentation generation system. The mechanism used to mark up source is similar to [doxygen](https://www.doxygen.nl/manual/docblocks.html). Namely
+
+```
+/**
+ ... text ... (JavaDoc style)
+ */
+void someFunctionA() {} 
+ 
+/*!
+ .. text .. (QT style)
+ another line
+ */ 
+void someFunctionB() {} 
+ 
+/// ... text ... (Multi line)
+/// another line
+void someFunctionC() {}
+ 
+//!... text ...  (QT Multi line)
+//! another line
+void someFunctionD() {}
+ 
+```
+ 
+All of the above examples will add the documentation for the declaration that appears after them. Also note that this slightly diverges from doxygen in that an empty line before and after in a multi line comment is *not* required.
+
+We can also document the parameters to a function similarly
+
+```
+/// My function
+void myFunction(
+    /// The A parameter 
+    int a, 
+    /// The B parameter
+    int b);
+```
+
+If you just need a single line comment to describe something, you can place the documentation after the parameter as in
+
+```
+
+/// My function
+void myFunction(    int a,      //< The A parameter     
+                    int b)      //< The B parameter
+{}
+```
+
+This same mechanisms work for other kinds of common situations such as with enums 
+
+```
+/// An enum
+enum AnEnum
+{
+    Value, ///< A value
+    /// Another value
+    /// With a multi-line comment
+    AnotherValue,
+};
+```
+
+Like `doxygen` we can also have multi line comments after a declaration for example
+
+```
+/// An enum
+enum AnEnum
+{
+    Value, ///< A value
+           ///< Some more information about `Value`
+           
+    /// Another value
+    /// With a multi-line comment
+    AnotherValue,
+};
+```
+
+
+
+
+To actually get Slang to output documentation you can use the `-doc` option from the `slangc` command line, or pass it in as parameter to `spProcessCommandLineArguments` or `processCommandLineArguments`. The documentation is currently output by default to the same `ISlangWriter` stream as diagnostics. So for `slangc` this will generally mean the terminal/stderr. 
+
+Currently the Slang doc system does not support any of the 'advanced' doxygen documentation features. If you add documentation to a declaration it is expected to be in [markdown](https://guides.github.com/features/mastering-markdown/). 
+
+Currently the only documentation style supported is a single file 'markdown' output. Future versions will support splitting into multiple files and linking between them. Also future versions may also support other documentation formats/standards.
+
+It is possible to generate documentation for Slangs internal `stdlib`. This can be achieved with `slangc` via
```

Comment:
Do we need to document this in the public docs? Ideally only devs on the project would care about how to document the stdlib.

Reply:
I kind of see this as documentation for ourselves too, and doesn't seem problematic that users at large can know this also. That said it probably isn't very important to users, so perhaps should be relegated to a more lowly location within the docs. Or perhaps it needs to go in another folder/and or file that isn't so user facing.